modules/ui/fields/address.js

```diff
@@ -55,7 +55,14 @@ export function uiFieldAddress(field, context) {
                     dist = geoSphericalDistance(choice.loc, l);
                 }
 
-                const value = resultProp && d.tags[resultProp] ? d.tags[resultProp] : d.tags.name;
+                const value = resultProp && d.tags[resultProp]
+                    ? d.tags[resultProp]
+                    : d.tags['name:en']
+                        ? d.tags['name:en'] // Fallback to English name
+                        : d.tags.name
+                            ? d.tags.name.split(';')[0].trim() // Use the first part of name (split by delimiter `;`)
+                            : null; // Default to null if no valid value exists
```

Comment:
I think this code would be clearer if you first refactor the line to use if/else statements instead of a ternary. Nested ternaries are very difficult to reason about. (Besides, without parentheses, I wouldn’t know at a glance whether the order of precedence is actually what you’ve implied using whitespace.)